sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/moduls/PrepareWrapperModule.java

```diff
@@ -1,17 +1,23 @@
 // SPDX-License-Identifier: MIT
 package com.mercedesbenz.sechub.wrapper.prepare.moduls;
 
-import java.util.List;
+import java.io.IOException;
 
 import org.springframework.stereotype.Service;
 
-import com.mercedesbenz.sechub.commons.model.SecHubConfigurationModel;
-import com.mercedesbenz.sechub.commons.model.SecHubRemoteDataConfiguration;
+import com.mercedesbenz.sechub.wrapper.prepare.prepare.PrepareWrapperContext;
 
 @Service
 public interface PrepareWrapperModule {
 
-    public boolean isAbleToPrepare(SecHubConfigurationModel model);
+    boolean isAbleToPrepare(PrepareWrapperContext context);
+
+    void prepare(PrepareWrapperContext context) throws IOException;
+
+    void cleanup(PrepareWrapperContext context) throws IOException;
+
+    boolean isDownloadSuccessful(PrepareWrapperContext context) throws IOException;
```

Comment:
Do really need this inside the interface?

From outside I would expect (psuedo code)
```
module are you enabled ?
no -> shake hands... 
yes -> module are you able to prepare it (context) ?
no -> okay shake hands...
yes -> prepare and throw an exception if it was not possible.
```
Even the "are you enabled" could be safed - could be done internally (but this okay)

But the `isDownloadSuccessful` ?